pkg/scan/process_image.go

```diff
@@ -271,21 +272,21 @@ func ScanFile(s *Scanner, f *os.File, iocs *[]output.IOCFound, layer string) err
 		}
 		err = yrScanner.ScanMem(buf)
 		if err != nil {
-			fmt.Println("Scan File Mmory Error", err)
+			logrus.Errorf("yara: %s: Error scanning file, error=%s", fileName, err.Error())
 			return filepath.SkipDir
 		}
 
 	}
 	var iocsFound []output.IOCFound
 	totalMatchesStringData := make([]string, 0)
 	for _, m := range matches {
-		matchesStringData := make([]string, len(m.Strings))
 		for _, str := range m.Strings {
-			if !strings.Contains(strings.Join(matchesStringData, " "), string(str.Data)) {
-				matchesStringData = append(matchesStringData, string(str.Data))
-				totalMatchesStringData = append(totalMatchesStringData, string(str.Data))
-			}
+			totalMatchesStringData = append(totalMatchesStringData, string(str.Data))
```

Comment:
Here we should be able to use:
```
func BytesToString(b []byte) (s string) {
	return unsafe.String(unsafe.SliceData(b), len(b))
}
```
If and only if `str.Data` is thrown away & never reused after that point

Reply:
Cool. This worked fine. Updated the code.